File: Problems/Python 3/997.py
class Solution:
    def findJudge(self, N: int, trust: List[List[int]]) -> int:
        trustNum = [0] * (N + 1)
        
        for trustee, trusted in trust:
            trustNum[trusted] += 1
            trustNum[trustee] -= 1
            
        for people in range(1, N + 1):
            if trustNum[people] == N - 1:
                return people
        
        return -1

File: Problems/C/997.c
int findJudge(int N, int** trust, int trustSize, int* trustColSize){
    int *trustNum = malloc(sizeof(int) * (N + 1));
    
    for(int i = 0; i < N + 1; i++)
        trustNum[i] = 0;
    
    for(int i = 0; i < trustSize; i++)
    {
        trustNum[trust[i][0]] -= 1;
        trustNum[trust[i][1]] += 1;
    }
    
    for(int i = 1; i < N + 1; i++)
        if(trustNum[i] == N - 1)
            return i;
    
    return -1;
}
